site/app/controllers/EmailStatusController.php

```diff
@@ -15,34 +16,80 @@ public function __construct(Core $core) {
     }
 
     /**
-     * @Route("/courses/{_semester}/{_course}/email_status")
+     * @Route("/courses/{_semester}/{_course}/email_status", methods={"GET"})
      * @AccessControl(role="INSTRUCTOR")
-     * @return MultiResponse
+     * @return WebResponse
      */
     public function getEmailStatusPage() {
         $semester = $this->core->getConfig()->getSemester();
         $course = $this->core->getConfig()->getCourse();
-        $result = $this->core->getQueries()->getEmailStatusWithCourse($semester, $course);
-        return MultiResponse::webOnlyResponse(
-            new WebResponse(
+
+        /** @phpstan-ignore-next-line */
+        $num_page = $this->core->getSubmittyEntityManager()->getRepository(EmailEntity::class)->getPageNum($semester, $course);
```

Comment:
Rather than having phpstan ignore this line you can type hint. You could do something like:

```suggestion
        /** @var EmailRepository $repository */
        $repository = $this->core->getSubmittyEntityManager()->getRepository(EmailEntity::class);
        $num_page = $repository->getPageNum($semester, $course);
```
You would need to add "use EmailRepository" here as well.